### PR TITLE
Added support for displaying time taken by each player & also Blitz mode Chess

### PR DESCRIPTION
**Pull request by kiran9k**
The time played by each player is being displayed along with the user name . Also , one can set a timeout ,  inorder to play in Blitz mode (to be specified in chesshub.js file)

### CODE REVIEW

**Comment by benas** [comment]
Cool! Thank you for this addition :+1: 

But I think your fork is not up to date with last commits in the master branch.
I have added some comments on outdated code parts. Please make sure to update your branch.

I'll merge your pull request asap.
Thank you for your contribution :-)
Cheers

File: .gitignore [hunk 1]
```diff
@@ -1,31 +0,0 @@
-# Logs
```
Comment (benas) [inline .gitignore]: can you undo the deletion of the .gitignore file? may be deleted by error?

File: views/layout.hbs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html ng-app="chesshubApp">
```
Comment (benas) [inline views/layout.hbs]: This is outdated comparing to master

File: views/layout.hbs [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
     <div class="footer">
         <p>&copy; ChessHub 2014, <i class="fa fa-code"></i> with <i class="fa fa-heart"></i> by
-            <a target="_blank" href="http://www.mahmoud-benhassine.fr">Mahmoud Ben Hassine</a></p>
+            <a target="_blank" href="http://www.mahmoudbenhassine.com">Mahmoud Ben Hassine</a></p>
```
Comment (benas) [inline views/layout.hbs]: This is outdated comparing to master
(Should be mahmoud-benhassine.fr, see 7f4b96a4641c8f01c67916c57cb7e0cde5c2bde9)

File: public/javascripts/chesshub.js [hunk 4]
```diff
@@ -133,6 +134,48 @@ $( document ).ready(function() {
         var side = $("#board").data('side');
         var opponentSide = side === "black" ? "white" : "black";
 
+
+        /*
```
Comment (benas) [inline public/javascripts/chesshub.js]: This is the heart of the pull request. Great! don't touch it :-)

File: app.js [hunk 5]
```diff
@@ -47,6 +47,12 @@ var login = require('./routes/login');
 var register = require('./routes/register');
 var search = require('./routes/search');
 
+app.use(function(req, res, next) {
+    res.header("Access-Control-Allow-Origin", "*");
+    res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    next();
```
Comment (benas) [inline app.js]: Please send this change in a separate pull request.
I have created an issue for that. see here: #8

File: app.js [hunk 6]
```diff
@@ -54,13 +60,24 @@ app.use('/account', account);
...
 app.use('/api', api);
 app.use('/search', search);
-
+app.set('trust proxy', '192.168.12.77'); // specify a single subnet
 // configure error handlers
 require('./config/errorHandlers.js')(app);
 
+
+
+var url_path='192.168.12.77';
```
Comment (benas) [inline app.js]: Host and port configuration is not related to adding time taken by each player.
Please send this change in a separate pull request.
I have created an issue for that. see here: #7

File: public/javascripts/chesshub.js [hunk 7]
```diff
@@ -205,7 +250,7 @@ $( document ).ready(function() {
          * When a new game is created, the game creator should wait for an opponent to join the game
          */
         socket.on('wait', function () {
-            var url = "http:/localhost:3000/game/" + token + "/" + opponentSide;
+            var url = "http:/192.168.12.77:3000/game/" + token + "/" + opponentSide;
```
Comment (benas) [inline public/javascripts/chesshub.js]: Should be fixed in issue #7

File: public/javascripts/chesshub.js [hunk 8]
```diff
@@ -364,7 +414,7 @@ $( document ).ready(function() {
         $( "#searchGameFormSubmit" ).on("click", function( event ) {
             $.ajax({
                 type: "POST",
-                url: "http://localhost:3000/search",
+                url: "http://192.168.12.77:3000/search",
```
Comment (benas) [inline public/javascripts/chesshub.js]: Should be fixed in issue #7

File: views/partials/game.hbs [hunk 9]
```diff
@@ -9,13 +9,15 @@
                 <h4 class="panel-title">Players</h4>
             </div>
             <div class="panel-body">
-                <p >
+                <p>
                     <img src="/images/chesspieces/wikipedia/bK.png" class="side-img" alt="Black"/>
                     <span id="player-black"></span> <span class="turn" id="turn-b">&nbsp;<i class=""></i></span>
+                    <span id="timeb" >{{time}}</span>
```
Comment (benas) [inline views/partials/game.hbs]: That's fine, to keep as is.

File: public/javascripts/chesshub.js [hunk 10]
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
     }
 
     // socket used for real time games
-    var socket = io('http://localhost:3000', { query: 'user=' + username });
+    var socket = io('http://192.168.12.77:3000', { query: 'user=' + username });
```
Comment (benas) [inline public/javascripts/chesshub.js]: Should be fixed in issue #7

File: public/javascripts/chesshub.js [hunk 11]
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
...
 
     //socket used to broadcast live games on tv page
-    var tvSocket = io('http://localhost:3000/tv');
+    var tvSocket = io('http://192.168.12.77:3000/tv');
```
Comment (benas) [inline public/javascripts/chesshub.js]: Should be fixed in issue #7

File: public/javascripts/chesshub.js [hunk 12]
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
...
 
     // socket used to broadcast events to monitoring page
-    var monitorSocket = io('http://localhost:3000/monitor');
+    var monitorSocket = io('http://192.168.12.77:3000/monitor');
```
Comment (benas) [inline public/javascripts/chesshub.js]: Should be fixed in issue #7

**Comment by benas** [comment]
A side note, you added handlebars-v2.0.0.js but you didn't imported it in the project. 
I'm not sure this is needed since handlebars is already imported:

https://github.com/benas/gamehub.io/blob/master/package.json#L19
https://github.com/benas/gamehub.io/blob/master/app.js#L28

That's all for my review.
Do not hesitate to add your comments. 
If it is ok for you, please update the pull request and I'll merge it asap. 
Your contribution is welcome!

**Comment by kiran9k** [comment]
Hi ,
I just updated the fork to the latest . 
I have added codes to display the time played by each player along with the user name . Also , support to Blitz mode chess is also provided .

Cheers

**Comment by benas** [comment]
Cool! this goes straight to the master branch :-)

Thank you for your contribution! :+1: 

Regards
Mahmoud
